File: 2237-longest-palindrome-by-concatenating-two-letter-words/2237-longest-palindrome-by-concatenating-two-letter-words.js
/**
 * @param {string[]} words
 * @return {number}
 */
var longestPalindrome = function(words) {
    var mpp=new Map()
    let result=0;
    for(const word of words){
        const rev=word[1]+word[0]
        if(mpp.get(rev)>0){
            result+=4;
            mpp.set(rev,mpp.get(rev)-1)

        }
        else{
            mpp.set(word,(mpp.get(word)||0)+1)
        }
    }
    for(const [word,count] of mpp.entries()){
        if(word[0]==word[1] && count>0){
            result+=2;
            break;
        }
    }
    return result;
};

File: 2237-longest-palindrome-by-concatenating-two-letter-words/2237-longest-palindrome-by-concatenating-two-letter-words.cpp
class Solution {
public:
    int longestPalindrome(vector<string>& words) {
        unordered_map<string,int> mpp;
        int result=0;
        for(const string&word:words){
            string rev=word;
            reverse(rev.begin(),rev.end());
            if(mpp[rev]){
                result+=4;
                mpp[rev]--;
            }
            else{
                mpp[word]++;
            }
        }
        for(auto it:mpp){
            if(it.first[0]==it.first[1] && it.second>0){
                result+=2;
                break;
            }
        }
        return result;
       

    }
};